src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -887,8 +964,24 @@ VideoDecoder::BatchDecodedOutput VideoDecoder::getFramesAtIndexes(
       throw std::runtime_error(
           "Invalid frame index=" + std::to_string(frameIndex));
     }
-    torch::Tensor frame = getFrameAtIndex(streamIndex, frameIndex).frame;
-    output.frames[i++] = frame;
+    int64_t pts = stream.allFrames[frameIndex].pts;
+    setCursorPtsInSeconds(ptsToSeconds(pts, stream.timeBase));
+    auto rawSingleOutput = getNextRawDecodedOutputNoDemux();
+    if (stream.filterState.sourceContext == nullptr) {
```

Comment:
It is not obvious to me that `stream.filterState.sourceContext` being null means that we should use sws_scale. It would be more obvious if the else branch explicitly used the value in this function, but the use is buried in the call to `convertFrameToTensorUsingFilterGraph()`.

I think we should store a `ColorConversionLibrary` value in the decoder's state, and use that value for control flow. We can always do not-null assertions as needed upon actual use.